src/midi_clock_tracker.rs

```diff
@@ -0,0 +1,49 @@
+use std::sync::mpsc::{self, Receiver, Sender};
+
+use crate::BeatNumber;
+
+const TICKS_PER_QUARTER_NOTE: u32 = 24;
+
+pub struct MidiClockTracker {
+    ticks_received: u32,
+    sender: Sender<BeatNumber>,
+}
+
+impl MidiClockTracker {
+    pub fn new() -> (Self, Receiver<BeatNumber>) {
+        let (sender, receiver) = mpsc::channel();
+
+        (
+            Self {
+                ticks_received: 0,
+                sender,
+            },
+            receiver,
+        )
+    }
+
+    pub fn tick(&mut self) -> () {
+        self.ticks_received += 1;
+        self.emit_beat_number();
```

Comment:
So this assumes that we're "starting from zero" and that the first MIDI clock tick received corresponds to the beginning of a measure. I'm not sure if/how there are ways to get richer information about what that clock tick actually means